### ISSUE TITLE
MediaElement crash when add new item to collectionview (Xamarin Forms- IOS) [Bug] 

### ISSUE THREAD

**Issue by Jearom** [xamarin/XamarinCommunityToolkit#670]
### Steps to Reproduce

1. Create Xamarin Forms Project
2. Create a Page with collectionView
3. Iniitialize with mutliple items (at least one media element)
4. Insert new item to collectionview 

### Expected Behavior
add New item to collectionview.

### Actual Behavior
Crash media element with error. 

Unhandled managed exception: Objective-C exception thrown.  Name: NSRangeException Reason: Cannot remove an observer <__XamarinObjectObserver 0x600001ed9900> for the key path "status" from <AVPlayerItem 0x60000171da40> because it is not registered as an observer.

### Native Crash Report

```
=================================================================
	Native Crash Reporting
=================================================================
Got a abrt while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x109a85e45 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_dump_native_crash_info
	0x109a7a98e - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_handle_native_crash
	0x109a8545f - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : sigabrt_signal_handler
	0x7fff5e7305fd - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libsystem_platform.dylib : _sigtramp
	0x10a3a2f90 - Unknown
	0x7fff200fab94 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libsystem_c.dylib : abort
	0x109c9875f - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : xamarin_unhandled_exception_handler
	0x109b34fb5 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_invoke_unhandled_exception_hook
	0x109a7a5aa - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_handle_exception_internal
	0x109a78a59 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_handle_exception
	0x1099fbee9 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_amd64_throw_exception
	0x10a4bb5b0 - Unknown
	0x109c984ab - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : xamarin_process_managed_exception
	0x10998dcaf - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : _ZL31native_to_managed_trampoline_36P11objc_objectP13objc_selectorPP11_MonoMethodj
	0x10999388c - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : -[__MonoMac_NSAsyncSynchronizationContextDispatcher xamarinApplySelector]
	0x7fff208532ba - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/Foundation.framework/Foundation : __NSThreadPerformPerform
	0x7fff2038c37a - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
	0x7fff2038c272 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : __CFRunLoopDoSource0
	0x7fff2038b754 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : __CFRunLoopDoSources0
	0x7fff20385f1f - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : __CFRunLoopRun
	0x7fff203856c6 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : CFRunLoopRunSpecific
	0x7fff2b76adb3 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices : GSEventRunModal
	0x7fff24675187 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : -[UIApplication _run]
	0x7fff2467a038 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : UIApplicationMain
	0x110ce2655 - Unknown
	0x11070b113 - Unknown
	0x10fb00a43 - Unknown
	0x109a8fb01 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_jit_runtime_invoke
	0x109ba8698 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_runtime_invoke_checked
	0x109badb15 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_runtime_exec_main_checked
	0x1099fb9bc - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : mono_jit_exec
	0x109ca8bdd - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : xamarin_main
	0x1099d3137 - /Users/barisyildirim/Library/Developer/CoreSimulator/Devices/4D6E538A-D7DF-4147-ACB0-F094BCCCD6D9/data/Containers/Bundle/Application/3B4D6922-501D-4900-899A-969EF2259176/mediainsidecollection.iOS.app/mediainsidecollection.iOS : main
	0x7fff20256409 - /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/libdyld.dylib : start
	0x3 - Unknown
```

### Basic Information

```
Microsoft Visual Studio Enterprise 2019
Version 16.8.2
VisualStudio.16.Release/16.8.2+30717.126
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Android Logcat Output   1.6.3
Android Logcat Output  -- Visual Studio Extension 

ASP.NET and Web Tools 2019   16.8.553.28003
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.553.28003
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.553.28003
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.8.553.28003
Azure Functions and Web Jobs Tools

C# Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

GitHub.VisualStudio   2.11.106.19330
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

GorillaPlayerPackage Extension   1.0
GorillaPlayerPackage Visual Studio Extension Detailed Info

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30924.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Mono Debugging for Visual Studio   16.8.43 (00471f8)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Progress Telerik UI for Xamarin Extension   2020.3.1117.1
Progress® Telerik® UI for Xamarin Extension

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62010.06180
Microsoft SQL Server Data Tools

TypeScript Tools   16.0.21016.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.8.0-beta.20507.4+da6be68280c89131cdba2045525b80890401defd
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.8.000.255 (d16-8@d002176)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.8.0.507 (remotes/origin/d16-8@e87b24884)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.8.112 (86385a3)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.1.0.17 (d16-8/c0e2b8e)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: be2226b
    Java.Interop: xamarin/java.interop/d16-8@79d9533
    ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
    SQLite: xamarin/sqlite/3.32.1@1a3276b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc


Xamarin.iOS and Xamarin.Mac SDK   14.4.1.3 (e30c41de3)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Workaround
I switch to Xamarin.Forms Media Element but it also has problems with Collectionview. 

### Reproduction Link

[CommunityMediaCollection.zip](https://github.com/xamarin/XamarinCommunityToolkit/files/5672058/CommunityMediaCollection.zip)

**Comment by AndreiMisiukevich** [comment]
Seems it duplicates #558 and it was fixed in #651 

Can you try the package published in #651?

**Comment by AndreiMisiukevich** [comment]
https://github.com/xamarin/XamarinCommunityToolkit/pull/651#issuecomment-739442077

**Comment by Jearom** [comment]
Hi @AndreiMisiukevich ,
    yes it fixes the issue but  xamarin.forms "5.0.0.1791-pre5" is necesssary.  Is there any chance to apply this fix in XF 4.8 ?

Thank you.

**Comment by AndreiMisiukevich** [comment]
> Hi @AndreiMisiukevich ,
> yes it fixes the issue but xamarin.forms "5.0.0.1791-pre5" is necesssary. Is there any way to apply this fix in XF 4.8 ?
> 
> Thank you.


I think, no. I guess XCT will require XF 5.0 

CC @jfversluis @jsuarezruiz

**Comment by jfversluis** [comment]
Yep we decided for now that we will depend on Forms 5. Even if we would decide otherwise, the controls will still need Forms 5 simply because Forms 5 has a lot of fixes that we depend on. Sorry about that :(

Closing this one as it seems fixed then!
